task/logging_abstraction/write_to_disk

BarAgam opened the pull request:
Added logging to disk functionality.
Added MultupleLogWrapper that delegates log calls to each log-wrapper it holds.
Initialized app's LogWrapper to be a MultipleLogWrapper that holds NativeLogWrapper and WriteToDiskLogWrapper.
Made a LogWrapperDecorator that filter log messages called VerbosityFilterLogWrapperDecorator.
WriteToDiskLogWrapper is wrapped inside a filter decorator to filter out info and verbose log messages.

barmako committed:
Done coding. Needs to be tested.

barmako committed:
VerbosityFilterLogWrapperDecorator - A decorator that filter log calls execution by verbosity levels. Instantiated with VerbosityConfiguration - a data class that determines which log verbosity levels to log.
Initialized WriteToDiskLogWrapper static log FileWriter at GGApplication

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/46" target="_blank"><img src="https://app.review.ninja/52010471/pull/46/badge" alt="ReviewNinja"/>

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/common/logging/WriteToDiskLogWrapper.java:
```diff
@@ -0,0 +1,218 @@
+        if (logsDir.mkdirs() || logsDir.isDirectory()) {
+            MediaScannerConnection.scanFile(context,
+                    new String[]{externalFilesDir.getAbsolutePath(), logsDir.getAbsolutePath()},
+                    null, null);
```
I think you should write a method to create the logger, and not the FileWriter and remove the FileWriter from the constructor

It shouldn't be modified outside of this class.

barmako committed:
Renaming:
*LoggerWrapper* => *Logger*

Much more readable and the wrapper was redundant anyway...

 Access modifiers for every Logger implementation has been changed to default access modifier (to allow access only at the same package)
  This way, the compiler can enforce that no concrete implementations are used outside the logging package.

barmako committed:
Refactored LoggerFactory for readability

barmako committed:
Merge remote-tracking branch 'origin/feature/logging_abstraction' into task/logging_abstraction/write_to_disk

# Conflicts:
#	app/src/main/java/com/teamagam/gimelgimel/app/view/BaseActivity.java
